quarkus/dist/src/main/content/bin/kc.sh

```diff
@@ -61,10 +64,16 @@ do
           break
           ;;
       *)
+          OPT=$(esceval "$1")
           case "$1" in
             start-dev) CONFIG_ARGS="$CONFIG_ARGS --profile=dev $1";;
-            -D*) SERVER_OPTS="$SERVER_OPTS $1";;
-            *) CONFIG_ARGS="$CONFIG_ARGS $1";;
+            -D*) SERVER_OPTS="$SERVER_OPTS ${OPT}";;
+            *) case "$1" in
+                 --optimized | --help | --help-all | -h) PRE_BUILD=false;;
```

Comment:
By the conclusion in #23071 we may not need to track --help, --help-all, nor -h in the shell scripts.  @mabartos WDYT?

Reply:
@mabartos Can we get your input on this? :)

Reply:
> By the conclusion in #23071 we may not need to track --help, --help-all, nor -h in the shell scripts, or for consistency should --show-config be added? @mabartos WDYT?

@shawkins @vmuzikar Yep, that's the question. Both variants might be a little bit controversial. It's true we do not need to track the `help` and `show-config` options in the script as they will not execute the augmentation phase due to this[1]. 

However, if they're not specified here, two processes are created. The first one, which is supposed to be for the 'pre-build', that will exit almost immediately, and then the second one, which is the application process itself. 

Comparison of both approaches:
1) Specify the options here
   **+** No additional process is created
   **-**  Worse maintainability

2) Do NOT specify the options here
   **-** Additional process is created
   **+**  Better maintainability

I'm kinda leaning towards option 2) for now, as these scripts are already quite messy. As all the `help` and `show-config` are terminal options, we do not need to care much about the performance. As I've mentioned, the process is quickly killed and does not consume resources in some big way. Thus, I prefer maintainability in this case.

**Future consideration:**
I don't like our execution scripts due to multiple reasons. The first one is described above, and the other is that I think these scripts should only manage the system stuff, not any comprehensive app logic. I'd even prefer to get rid of checking the `build`, `--optimized`, `kc.config.build-and-exit`, `kc.config.built` options in these scripts and rather move the handling them to the app logic. 

But we can discuss details elsewhere as it's out of the scope of this PR.    

[1] https://github.com/keycloak/keycloak/blob/a7d082e390ca128a44ad9e6ec6f5b8d2bfacfd0b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java#L130